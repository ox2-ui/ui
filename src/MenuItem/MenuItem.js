import React, { PropTypes } from 'react';
import Thumb from '../Thumb';

/**
 * MenuItem Component
 */
const MenuItem = ({ value, title, className, style, onClick, hidePicture, image, icon, pictureWidth, pictureHeight, pictureBgColor, iconColor, textColor }) => (
  <div className={className} style={style}>
    <div onClick={() => onClick(value)} className="display:flex flex:items-center color:white padding:5 select:none cursor:pointer">
      { !hidePicture &&
        <Thumb
          image={image}
          icon={icon || 'cactus'}
          width={pictureWidth | 28}
          height={pictureHeight | 28}
          color={pictureBgColor}
          iconColor={iconColor}
        />
      }
      {/* class min-w:0 added to fix Flex child truncate issue */}
      <div className="display:flex flex:column flex:content-center margin-l:5 min-w:0 line-h:130p font:roboto-condensed">
        <div className={`select:text size:16 line-h:120p text:truncate text-color:${textColor || 'backdrop-alt'}`}>{title}</div>
      </div>
    </div>
  </div>
);


MenuItem.propTypes = {
  className: PropTypes.string,
  hidePicture: PropTypes.bool,
  icon: PropTypes.string,
  iconColor: PropTypes.string,
  image: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  pictureBgColor: PropTypes.string,
  pictureHeight: PropTypes.number,
  pictureWidth: PropTypes.number,
  style: PropTypes.object,
  title: PropTypes.string,
  value: PropTypes.string.isRequired,
};

export default MenuItem;
