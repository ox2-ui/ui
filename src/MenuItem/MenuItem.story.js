import React from 'react';
import { storiesOf } from '@kadira/storybook';
import MenuItem from './MenuItem';

const handler = () => console.log('🍇'); // eslint-disable-line no-console

storiesOf('MenuItem', module)
  .add('default', () => (
    <MenuItem
      value={'/system/status'}
      title={'Settings'}
      icon={'cog'}
      iconColor={'brand'}
      pictureBgColor={'white'}
      onClick={handler}
    />
  ));
