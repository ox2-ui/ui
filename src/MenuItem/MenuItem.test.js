/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import MenuItem from './MenuItem';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <MenuItem
      value={'/system/status'}
      title={'Settings'}
      icon={'cog'}
      iconColor={'brand'}
      pictureBgColor={'white'}
      onClick={handler}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
