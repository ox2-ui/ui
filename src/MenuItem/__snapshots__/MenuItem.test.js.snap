exports[`test renders correctly 1`] = `
<MenuItem
  icon="cog"
  iconColor="brand"
  onClick={[Function]}
  pictureBgColor="white"
  subtitle1="Ox2 Limited"
  subtitle2="Marketing Manager"
  title="Settings"
  value="/system/status">
  <div>
    <div
      className="display:flex flex:items-center color:white padding:5 select:none cursor:pointer"
      onClick={[Function]}>
      <Thumb
        color="white"
        height={28}
        icon="cog"
        iconColor="brand"
        width={28}>
        <div
          className="display:flex flex:items-center flex:content-center border:rounded color:white"
          style={
            Object {
              "WebkitFlex": "0 0 28px",
              "flex": "0 0 28px",
              "height": "28px",
              "width": "28px",
            }
          }>
          <Icon
            className="text-color:brand"
            icon="cog"
            size="19.6px">
            <span
              className="icon-cog text-color:brand"
              style={
                Object {
                  "color": undefined,
                  "fontSize": "19.6px",
                }
              } />
          </Icon>
        </div>
      </Thumb>
      <div
        className="display:flex flex:column flex:content-center margin-l:5 min-w:0 line-h:130p font:roboto-condensed">
        <div
          className="select:text size:16 line-h:120p text:truncate text-color:backdrop-alt">
          Settings
        </div>
      </div>
    </div>
  </div>
</MenuItem>
`;
